client/src/commands/getTcb.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as vscode from 'vscode';
+
+import * as lspext from '../../common/lsp_extensions';
+import {AngularLanguageClient} from '../client';
+
+const TCB_HIGHLIGHT_DECORATION = vscode.window.createTextEditorDecorationType({
+  backgroundColor: new vscode.ThemeColor('editor.selectionHighlightBackground'),
+});
+
+class SourceMap {
+  private readonly lines: Array<readonly[/* line offset */ number, /* line length */ number]>;
+  constructor(source: string) {
+    let offset = 0;
+    this.lines = source.split('\n').map(line => {
+      const lineLen = line.length + 1 /* ='\n' */;
+      const offsetAndLen = [offset, lineLen] as const;
```

Comment:
Small nit on naming: my brain reads these as `lineLength` and `offsetLength` but it takes longer because it has to always do the conversion. I would personally prefer the words spelled out but I'll survive if you keep it as-is :)

Reply:
I am not sure I follow, but I tried to rename the variables to spell everything out. Can you let me know if it is better?